### ISSUE TITLE
ArgumentException in MSBuildProjectLoader.Worker.cs line 273 (duplicate refs)

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/roslyn#31541]
[DuplicateReferences.zip](https://github.com/dotnet/roslyn/files/2646711/DuplicateReferences.zip)

1. Unzip the attached DuplicateReferences.zip
2. Open the .sln in MSBuildWorkspace (an easy way to do that is using https://github.com/KirillOsenkov/SourceBrowser):
  2.1. git clone https://github.com/KirillOsenkov/SourceBrowser
  2.2. In HtmlGenerator project properties, change the first argument to the path to the .sln file you unzipped
  2.3. Enable first-chance exceptions and F5

Exception:
```
 	System.ThrowHelper.ThrowArgumentException Line 72	C#
 	System.Collections.Generic.Dictionary<string, System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.ProjectInfo>>.Insert Line 338	C#
 	System.Collections.Generic.Dictionary<System.__Canon, System.Collections.Immutable.ImmutableArray<System.__Canon>>.Add Line 193	C#
>	Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.LoadProjectInfosFromPathAsync Line 275	C#
 	Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.TryLoadAndAddReferenceAsync Line 221	C#
 	Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.ResolveReferencesAsync Line 204	C#
 	[Resuming Async Method]	
 	Microsoft.CodeAnalysis.MSBuild.MSBuildProjectLoader.Worker.VerifyUnloadableProjectOutputExistsAsync Line 277	C#
 	[Resuming Async Method]	
 	Microsoft.CodeAnalysis.MSBuild.Build.ProjectBuildManager.TryGetOutputFilePathAsync Line 157	C#
 	[Resuming Async Method]	
 	Microsoft.CodeAnalysis.MSBuild.Build.ProjectBuildManager.LoadProjectAsync Line 148	C#
```

It seems that the key is that the referenced project has the .qproj extension. I'm not sure if it has to be multi-targeted and whether the duplicate reference has to be added via a Target that runs at runtime.

See potentially related:
https://github.com/dotnet/roslyn/commit/7e4ba1dfcac89f719efc2265f14822b775555802
https://github.com/dotnet/roslyn/issues/30382
https://github.com/dotnet/roslyn/issues/8408
https://github.com/dotnet/roslyn/issues/31390
https://github.com/dotnet/roslyn/issues/12356
https://github.com/dotnet/roslyn/issues/12101

Actual place where this happens:
https://github.com/dotnet/corefx/blob/bac5bb8a23b5036cb742fef762bcb3a05afb8979/eng/referenceFromRuntime.targets#L19-L25

Note that SourceBrowser already does deduplication of references here:
https://github.com/KirillOsenkov/SourceBrowser/blob/6f61d387b7a1c977815556f493ceedfc5418366a/src/HtmlGenerator/Pass1-Generation/SolutionGenerator.cs#L277

However in this case it doesn't take effect because the .qproj is not part of the .sln (I think).

**Comment by DustinCampbell** [comment]
Thanks @KirillOsenkov, I'll take a look.

**Comment by DustinCampbell** [comment]
@KirillOsenkov: Is it intended that this repro doesn't build? It looks like there's a syntax error in Library.qproj (using a coma instead of a semi-colon in `<TargetFrameworks>`)

**Comment by DustinCampbell** [comment]
This is a duplicate of https://github.com/dotnet/roslyn/issues/30382, which was fixed in Roslyn 2.11. You can grab the latest bits from https://dotnet.myget.org/gallery/roslyn to get the fix.

**Comment by KirillOsenkov** [comment]
Ah, yes, nice, that will fix it. 

No, the comma in TargetFrameworks was not intentional, it's me not remembering the syntax.

Thanks.
